queryscript/src/compile/generics.rs

```diff
@@ -201,8 +205,28 @@ impl Generic for SumGeneric {
         })?;
         Ok(())
     }
+
+    fn resolve(&self) -> Result<CRef<MType>> {
+        let this = self.clone();
+        this.0.clone().then(move |_: Ref<MType>| {
+            let rt = match this.to_runtime_type().context(RuntimeSnafu {
+                loc: ErrorLocation::Unknown,
+            }) {
+                Ok(rt) => rt,
+                Err(_) => {
+                    return Ok(mkcref(MType::Generic(Located::new(
+                        Arc::new(this.clone()),
+                        SourceLocation::Unknown,
+                    ))))
+                }
+            };
+
+            Ok(mkcref(MType::from_runtime_type(&rt)?))
```

Comment:
why not just return the original cref here? Seems like we copy the whole thing twice (once to check `to_runtime_type()` and another to reconstruct a cref?)

Reply:
Which original cref?  The goal is to return the atomic type the generic wants to turn into (i.e. the output of `to_runtime_type()`).

Reply:
Yeah I get it now